fqdn/__init__.py

```diff
@@ -143,3 +143,38 @@ def __eq__(self, other):
 
     def __hash__(self):
         return hash(self.absolute) + hash("fqdn")
+
+
+class Label:
+    """
+    A RFC 1123 label must consist of alphanumeric characters
+    or '-', and must start and end with an alphanumeric character
+    for example 'my-name',  or '123-abc'
+    """
+
+    RFC_1123_LABLE_REGEX = r"[a-zA-Z0-9]([-a-zA-Z0-9]*[a-zA-Z0-9])?$"
```

Comment:
There is a little typo in your variable naming:
```suggestion
    RFC_1123_LABEL_REGEX = r"[a-zA-Z0-9]([-a-zA-Z0-9]*[a-zA-Z0-9])?$"
```